### PR TITLE
Feature/mongo db europe2018

### PR DESCRIPTION
**Pull request by Nxtra**

### CODE REVIEW

File: _posts/2018-11-06-mongodb-europe-018.md [hunk 1]
```diff
@@ -0,0 +1,404 @@
...
+    "server.name" : "db-01.example.com",
+    "server.port" : "1433"
+  }
+}
+```
+
+They use Avro for the schema definition in combination with a schema-registry.
+Interested clients can then read the data of the topics and do their single-view-thing on the data and save it to MongoDB.
+The view is being build up, message per message.
+The view in  MongoDB is then pushed back to Kafka as another stream to provide this data to interested parties.
```
Comment (tomvdbulck) [inline _posts/2018-11-06-mongodb-europe-018.md]:
```suggestion
Afterwards this view in  MongoDB is then pushed back to Kafka as another stream to provide this data to interested parties.
```

File: _posts/2018-11-06-mongodb-europe-018.md [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+
+Currently, if you want to visualize data in a MongoDB database, you either have to code it yourself or rely on a 3rd party tool and migrate your data to a different platform (for example: Kibana with Elasticsearch is a popular tool).
+Needless to say that this can be quite cumbersome.
+MongoDB Charts intends to solve this.
+
+So what is it? MongoDB Charts is a visualization tool that runs as a standalone webserver, you access it via a web browser.
+
+In Charts, you define data sources by pointing to a collection or view on a MongoDB instance.
+Then you can create all kinds of visualizations based on the data source, using various aggregation and filter functions.
+Finally, you can combine charts into dashboards with a customized layout, and you can share them with other users.
```
Comment (tomvdbulck) [inline _posts/2018-11-06-mongodb-europe-018.md]:
```suggestion
Finally, you can combine charts into dashboards with a customized layout and you can share them with other users.
```

File: _posts/2018-11-06-mongodb-europe-018.md [hunk 3]
```diff
@@ -0,0 +1,404 @@
...
+
+A picture is worth a thousand words, so to give you a better idea of what it is all about, let's look at an animation of Charts in action:
+
+<div style="text-align: center;">
+  <img src="/img/2018-11-08-mongodb-europe-2018/mongodb-charts-01.gif" width="100%">
+</div>
+
+Charts is still in beta, but you can already try it out. MongoDB provides a docker image that you can [download](https://www.mongodb.com/download-center/charts) via the MongoDB download center.
```
Comment (tomvdbulck) [inline _posts/2018-11-06-mongodb-europe-018.md]:
```suggestion
Charts is still in beta but you can already try it out. 
MongoDB provides a docker image that you can [download](https://www.mongodb.com/download-center/charts) via the MongoDB download center.
```

File: _posts/2018-11-06-mongodb-europe-018.md [hunk 4]
```diff
@@ -0,0 +1,404 @@
...
+<div style="text-align: center;">
+  <img src="/img/2018-11-08-mongodb-europe-2018/create-mongodb-atlas-cluster.png" width="100%">
+</div>
+
+Start your own cluster: [https://cloud.mongodb.com](https://cloud.mongodb.com){:target="_blank" rel="noopener noreferrer"}
+
+### Cloud
+MongoDB Atlas is a cross-platform database. 
+You can run it on Amazon Web Services, Google Cloud or Microsoft Azure. 
+It provides you an availability map that shows you which users are served from where an what latency should be expected.
```
[the remark this answers is not in the document]
Comment (ChrisDeBruyne) [reply _posts/2018-11-06-mongodb-europe-018.md]: done

File: _posts/2018-11-06-mongodb-europe-018.md [hunk 5]
```diff
@@ -0,0 +1,404 @@
...
+title: "MongoDB Europe 2018"
+image: /img/2018-11-08-mongodb-europe-2018/main-image-mdbe.png
+tags: [Development,MongoDB,DBA,Data,Kubernetes,Conference]
+category: Development
+comments: true
+---
+
+> MongoDB Europe is a yearly conference where MongoDB shows of their latest features and new products.
+> This year the venue took place in Old Billingsgate Walk, London.
+> This blog post summarizes several of the given sessions.
```
[the remark this answers is not in the document]
Comment (ChrisDeBruyne) [reply _posts/2018-11-06-mongodb-europe-018.md]: done

File: _posts/2018-11-06-mongodb-europe-018.md [hunk 6]
```diff
@@ -0,0 +1,408 @@
...
+It provides you an availability map that shows you which users are served from where an what latency should be expected.
+<div style="text-align: center;">
+  <img src="/img/2018-11-08-mongodb-europe-2018/global-clusters.png" width="100%">
+</div>
+
+### Scalability
+When your database grows it is easy to scale up or out.
+You can scale up by increasing the size of the instance on which your database runs.
+Scaling out is done by the process of sharding.
+Here we are storing data of the same collection across multiple Replica Sets to decrease the size of the partition and increase read-write performance.
```
Comment (tomvdbulck) [inline _posts/2018-11-06-mongodb-europe-018.md]:
```suggestion
Here we are storing data of the same collection across multiple smaller partitions so we can distribute these partitions over multiple machines and increase read-write performance.
```

File: _posts/2018-11-06-mongodb-europe-018.md [hunk 7]
```diff
@@ -0,0 +1,408 @@
...
+This way you do not run into the limitations of a single server.
+
+To ensure that MongoDB stores data equally across shards you need the right strategy of choosing a partition key.
+
+### High availability 
+To ensure maximum uptime the procedure to recover from instance failures is completely automated.
+When a primary node goes down, a new primary is chosen immediately by the system of voting.
+All nodes vote on who should become the new primary. 
+The node with the majority of the votes becomes the new primary.
+A general guideline is to have a replica set that consists of one primary node and at least two secondary nodes.
```
Comment (tomvdbulck) [inline _posts/2018-11-06-mongodb-europe-018.md]: move this line as first under ### High availabiltiy

File: _posts/2018-11-06-mongodb-europe-018.md [hunk 8]
```diff
@@ -0,0 +1,408 @@
...
+* Watch out for frameworks generating queries.  
+   They can create bad performing aggregations or queries.  
+   For them it was better to write some queries explicitly.  
+* Use custom _id for unique determination of your model, it saves an index and RAM
+* Watch out for unbounded document growth.
+
+
+# [MongoDB Enterprise Operator for Kubernetes at Amadeus](https://sched.co/FmAc){:target="_blank" rel="noopener noreferrer"}
+
+Amadeus is the world's largest technology company dedicated to the travel industry.
```
Comment (tomvdbulck) [inline _posts/2018-11-06-mongodb-europe-018.md]:
```suggestion
[Amadeus](https://amadeus.com/en) is the world's largest technology company dedicated to the travel industry.
```

File: _posts/2018-11-06-mongodb-europe-018.md [hunk 9]
```diff
@@ -0,0 +1,408 @@
...
+They have developed an impressive MongoDB farm - a large environment with 100 clusters, some of which run more than 100 shards, some of which run 100TB MongoDB databases.
```
Comment (tomvdbulck) [inline _posts/2018-11-06-mongodb-europe-018.md]:
```suggestion
They have developed an impressive MongoDB farm, a large environment with 100 clusters.
Some of these clusters run more than 100 shards, while others have 100TB MongoDB databases.
```

File: _posts/2018-11-06-mongodb-europe-018.md [hunk 10]
```diff
@@ -0,0 +1,408 @@
...
+Amadeus processes more than 1 trillion availability requests per day. 
+For each single search you do on a website they receive 500.000 availability requests.
```
Comment (tomvdbulck) [inline _posts/2018-11-06-mongodb-europe-018.md]: hier mis ik wat context.
Is dat het zoeken op een boekingssite of iets van een reisbureau?

File: _posts/2018-11-06-mongodb-europe-018.md [hunk 11]
```diff
@@ -0,0 +1,408 @@
...
+So search responsibly ;-)
+The number of requests per day grows by 50% each year.
+Worst day is january 2th due to new years resolutions!
```
Comment (tomvdbulck) [inline _posts/2018-11-06-mongodb-europe-018.md]:
```suggestion
The second of January has by far the most requests, due to new years resolutions!
```

File: _posts/2018-11-06-mongodb-europe-018.md [hunk 12]
```diff
@@ -0,0 +1,408 @@
...
+If this day is in the weekend, all systems are pushed to their limits.
```
Comment (tomvdbulck) [inline _posts/2018-11-06-mongodb-europe-018.md]:
```suggestion
If this day is in the weekend all systems are pushed to their limits.
```

File: _posts/2018-11-06-mongodb-europe-018.md [hunk 13]
```diff
@@ -0,0 +1,408 @@
...
+The fair database for one of their big clients, Kayak, is 100 Tb in size and changes daily.
```
Comment (tomvdbulck) [inline _posts/2018-11-06-mongodb-europe-018.md]:
```suggestion
The fair database for one of their big clients, Kayak, is 100TB in size and changes daily.
```

File: _posts/2018-11-06-mongodb-europe-018.md [hunk 14]
```diff
@@ -0,0 +1,408 @@
...
+	"birthYear" : 1996
+},
+{
+	"_id" : ObjectId("5be40f6e7047ead15753d074"),
+	"firstName" : "Nick",
+	"lastName" : "Van Hoof",
+	"birthYear" : 1992
+}
+```
+A person has the fields lastName, firstName and birthYear (and of course for some people more info could be stored).
```
[the remark this answers is not in the document]
Comment (Nxtra) [reply _posts/2018-11-06-mongodb-europe-018.md]: Here I am hinting on the fact that there is no fixed database schema

File: _posts/2018-11-06-mongodb-europe-018.md [hunk 15]
```diff
@@ -0,0 +1,408 @@
...
+So search responsibly ;-)
+The number of requests per day grows by 50% each year.
+Worst day is january 2th due to new years resolutions!
+If this day is in the weekend, all systems are pushed to their limits.
+The fair database for one of their big clients, Kayak, is 100 Tb in size and changes daily.
+That's some pretty big numbers there.
+No wonder that Amadeus is a happy user of the MongoDB Enterprise Operator for Kubernetes.
+
+Starting with the MongoDB Ops Manager v4.0, MongoDB officially supports the management and deployment of MongoDB in Kubernetes with Backup, Automation, Alerting and Monitoring.
+An operator has app-specific awareness about stateful applications, so it knows how to deploy them.
```
Comment (tomvdbulck) [inline _posts/2018-11-06-mongodb-europe-018.md]:
```suggestion
A [MongoDB Enterprise Kubernetes Operator](https://github.com/mongodb/mongodb-enterprise-kubernetes) has app-specific awareness about stateful applications, so it knows how to deploy them.
```

File: _posts/2018-11-06-mongodb-europe-018.md [hunk 16]
```diff
@@ -0,0 +1,408 @@
...
+This operator helps automating scripted tasks and enables MongoDB-as-a-service for developers.
+This operator talks to Ops Manager and delegates the creation of clusters, shards, backups and automation to Ops Manager.
```
Comment (tomvdbulck) [inline _posts/2018-11-06-mongodb-europe-018.md]:
```suggestion
It talks to Ops Manager and delegates the creation of clusters, shards, backups and automation to Ops Manager.
```

File: _posts/2018-11-06-mongodb-europe-018.md [hunk 17]
```diff
@@ -0,0 +1,408 @@
...
+
+Scale out or back in with a simple change in the config yaml and 
+```bash
+cubectl apply -f file.yaml
+```
+
+# MongoDB Charts
+- [Bringing Data to Life with MongoDB Charts](https://sched.co/FtWP), Guillaume Meister.
+
+Currently, if you want to visualize data in a MongoDB database, you either have to code it yourself or rely on a 3rd party tool and migrate your data to a different platform (for example: Kibana with Elasticsearch is a popular tool).
```
Comment (tomvdbulck) [inline _posts/2018-11-06-mongodb-europe-018.md]:
```suggestion
Currently, if you want to visualize data in a MongoDB database you either have to code it yourself, or rely on a 3rd party tool and migrate your data to a different platform (for example: Kibana with Elasticsearch is a popular tool).
```

File: _posts/2018-11-06-mongodb-europe-018.md [hunk 18]
```diff
@@ -0,0 +1,408 @@
...
+Needless to say that this can be quite cumbersome.
+MongoDB Charts intends to solve this.
```
Comment (tomvdbulck) [inline _posts/2018-11-06-mongodb-europe-018.md]:
```suggestion
[MongoDB Charts](https://www.mongodb.com/products/charts) intends to solve this.
```

File: _posts/2018-11-06-mongodb-europe-018.md [hunk 19]
```diff
@@ -0,0 +1,408 @@
...
+
+So what is it? MongoDB Charts is a visualization tool that runs as a standalone webserver, you access it via a web browser.
```
Comment (tomvdbulck) [inline _posts/2018-11-06-mongodb-europe-018.md]:
```suggestion
So what is it? 
MongoDB Charts is a visualization tool that runs as a standalone webserver, you access it via a web browser.
```

File: _posts/2018-11-06-mongodb-europe-018.md [hunk 20]
```diff
@@ -0,0 +1,408 @@
...
+
+In Charts, you define data sources by pointing to a collection or view on a MongoDB instance.
```
Comment (tomvdbulck) [inline _posts/2018-11-06-mongodb-europe-018.md]:
```suggestion
In Charts you define data sources by pointing to a collection or view on a MongoDB instance.
```

File: _posts/2018-11-06-mongodb-europe-018.md [hunk 21]
```diff
@@ -0,0 +1,408 @@
...
+Then you can create all kinds of visualizations based on the data source, using various aggregation and filter functions.
+Finally, you can combine charts into dashboards with a customized layout and you can share them with other users.
```
Comment (tomvdbulck) [inline _posts/2018-11-06-mongodb-europe-018.md]:
```suggestion
Finally, you can combine charts into dashboards with a customized layout and you can share these with other users.
```

File: _posts/2018-11-06-mongodb-europe-018.md (hunk 10 above)
Comment (ChrisDeBruyne) [reply _posts/2018-11-06-mongodb-europe-018.md]: voldoende als ik gewoon 'flight' ertussen zet

> Amadeus processes more than 1 trillion flight availability requests per day.

Comment (tomvdbulck) [reply _posts/2018-11-06-mongodb-europe-018.md]: ja, dat maakt het idd mss wel wat duidelijker

File: _posts/2018-11-20-mongodb-europe-018.md [hunk 22]
```diff
@@ -0,0 +1,411 @@
...
+6. [Meet the experts](#meet-the-experts)
+7. [Streaming data pipelines with MongoDB and Kafka at AO](#streaming-data-pipelines-with-mongodb-and-kafka-at-ao)
+8. [MongoDB Enterprise Operator for Kubernetes at Amadeus](#mongodb-enterprise-operator-for-kubernetes-at-amadeus)
+9. [MongoDB Charts](#mongodb-charts)
+10. [Closing](#closing)
+
+# Atlas
+([MongoDB Atlas for your Enterprise](https://sched.co/FmAF){:target="_blank" rel="noopener noreferrer"}, Vladislava Stevanovic & Veronica Tudor)
+
+With Atlas, MongoDB brings us DBaaS (Database As A Service).
```
Comment (TimDG) [inline _posts/2018-11-20-mongodb-europe-018.md]: This section reads a bit odd as the sentences are all structured identically: someone verbed something.
Try to shuffle it around a bit for a better reading experience.

File: _posts/2018-11-20-mongodb-europe-018.md [hunk 23]
```diff
@@ -0,0 +1,411 @@
...
+A cluster can be started for free in just a few clicks.
+
+<div style="text-align: center;">
+  <img src="/img/2018-11-08-mongodb-europe-2018/create-mongodb-atlas-cluster.png" width="100%">
+</div>
+
+Start your own cluster: [https://cloud.mongodb.com](https://cloud.mongodb.com){:target="_blank" rel="noopener noreferrer"}
+
+### Cloud
+MongoDB Atlas is a cross-platform database.
```
Comment (TimDG) [inline _posts/2018-11-20-mongodb-europe-018.md]: Combine this sentence and the next (perhaps using `which`)

File: _posts/2018-11-20-mongodb-europe-018.md [hunk 24]
```diff
@@ -0,0 +1,411 @@
...
+<div style="text-align: center;">
+  <img src="/img/2018-11-08-mongodb-europe-2018/create-mongodb-atlas-cluster.png" width="100%">
+</div>
+
+Start your own cluster: [https://cloud.mongodb.com](https://cloud.mongodb.com){:target="_blank" rel="noopener noreferrer"}
+
+### Cloud
+MongoDB Atlas is a cross-platform database. 
+You can run it on Amazon Web Services, Google Cloud or Microsoft Azure. 
+It provides you an availability map that shows you which users are served from where an what latency should be expected.
```
Comment (TimDG) [inline _posts/2018-11-20-mongodb-europe-018.md]: an**d** what the expected latency is.

File: _posts/2018-11-20-mongodb-europe-018.md [hunk 25]
```diff
@@ -0,0 +1,411 @@
...
+</div>
+
+### Scalability
+When your database grows it is easy to scale up or out.
+You can scale up by increasing the size of the instance on which your database runs.
+Scaling out is done by the process of sharding.
+Here we are storing data of the same collection across multiple smaller partitions so we can distribute these partitions over multiple machines and increase read-write performance.
+This way you do not run into the limitations of a single server.
+
+To ensure that MongoDB stores data equally across shards you need the right strategy of choosing a partition key.
```
Comment (TimDG) [inline _posts/2018-11-20-mongodb-europe-018.md]: across shards,

File: _posts/2018-11-20-mongodb-europe-018.md [hunk 26]
```diff
@@ -0,0 +1,411 @@
...
+
+### High availability 
+When a primary node goes down, a new primary is chosen immediately by the system of voting.
```
Comment (TimDG) [inline _posts/2018-11-20-mongodb-europe-018.md]: "by the system of voting?" Shouldn't we use an indefinite article here?

File: _posts/2018-11-20-mongodb-europe-018.md [hunk 27]
```diff
@@ -0,0 +1,411 @@
...
+All nodes vote on who should become the new primary. 
+The node with the majority of the votes becomes the new primary.
```
Comment (TimDG) [inline _posts/2018-11-20-mongodb-europe-018.md]: "the majority of votes"

File: _posts/2018-11-20-mongodb-europe-018.md [hunk 28]
```diff
@@ -0,0 +1,411 @@
...
+A general guideline is to have a replica set that consists of one primary node and at least two secondary nodes.
+To ensure maximum uptime the procedure to recover from instance failures is completely automated.
+
+
+# MongoDB University
+(General Session Keynote)
```
Comment (TimDG) [inline _posts/2018-11-20-mongodb-europe-018.md]: Why is this here?

File: _posts/2018-11-20-mongodb-europe-018.md [hunk 29]
```diff
@@ -0,0 +1,411 @@
...
+
+### High availability 
+When a primary node goes down, a new primary is chosen immediately by the system of voting.
+All nodes vote on who should become the new primary. 
+The node with the majority of the votes becomes the new primary.
+A general guideline is to have a replica set that consists of one primary node and at least two secondary nodes.
+To ensure maximum uptime the procedure to recover from instance failures is completely automated.
+
+
+# MongoDB University
```
Comment (TimDG) [inline _posts/2018-11-20-mongodb-europe-018.md]: Reclameblok?

File: _posts/2018-11-20-mongodb-europe-018.md [hunk 30]
```diff
@@ -0,0 +1,411 @@
...
+I speak from my own experience when I say that the University platform is great to work with, and the courses are very well taught.
+Find all available resources here: [https://university.mongodb.com/](https://university.mongodb.com/){:target="_blank" rel="noopener noreferrer"}.
+
+
+
+# Compass Aggregation Pipeline Builder
+(General Session Keynote)
+
+Recently the aggregation pipeline builder was introduced in [MongoDB Compass](https://www.mongodb.com/products/compass){:target="_blank" rel="noopener noreferrer"}.
+This allows you to create an aggregation pipeline step by step.
```
Comment (TimDG) [inline _posts/2018-11-20-mongodb-europe-018.md]: Combine with the previous sentence into 1

File: _posts/2018-11-20-mongodb-europe-018.md [hunk 31]
```diff
@@ -0,0 +1,411 @@
...
+And that makes it easy to debug the pipeline along the way.
+Let's see an example:   
+Suppose I have a collection which contains documents that represent a person.
+Here are two examples of elements in the collections:
```
Comment (TimDG) [inline _posts/2018-11-20-mongodb-europe-018.md]: This sentence feels forced. Why not add ", like this:" to end of the previous one?

File: _posts/2018-11-20-mongodb-europe-018.md [hunk 32]
```diff
@@ -0,0 +1,411 @@
...
+	"firstName" : "Nick",
+	"lastName" : "Van Hoof",
+	"birthYear" : 1992
+}
+```
+A person has the fields lastName, firstName and birthYear (and of course for some people more info could be stored).
+I want to build a pipeline with the following functionality:
+* I want to filter out all people that share my lastname "Van Hoof"
+* Then I want to count how many times these people also share the same firstname and birthyear
+* Next I want to group them by birthYear so that I can see how many people were named "Nick van Hoof " (my fullname) in 1992.
```
Comment (TimDG) [inline _posts/2018-11-20-mongodb-europe-018.md]:
```suggestion
* Next I want to group them by birthYear so that I can see how many people were named "Nick Van Hoof " (my fullname) in 1992.
```

File: _posts/2018-11-20-mongodb-europe-018.md [hunk 33]
```diff
@@ -0,0 +1,411 @@
...
+This happens when you use an operator that needs all the data before producing results, so results can't be streamed.
+The most common culprits are the aggregation operators such as `$sort`, `$group`, `$bucket`, `$count` and `$facet`.
+
+Possible solutions:
+- Create a compound index to support your query and make sure that the sort order in the index is the same as in your query.
+- Offload the query to a secondary member.
+- Work around the issue by using a precalculated count.
+
+### Problem 2: `$and` is fast, `$or` is slow
+Sometimes a query is fast when you use the `$and` operator but slow when you use the `$or` operator.
```
Comment (TimDG) [inline _posts/2018-11-20-mongodb-europe-018.md]: Why? Or is this common knowledge?

File: _posts/2018-11-20-mongodb-europe-018.md [hunk 34]
```diff
@@ -0,0 +1,411 @@
...
+* Train your people.
+* Work smarter, not harder!
+
+# Stitch
+([Ch-Ch-Ch-Ch-Changes: Taking Your MongoDB Stitch Application to the Next Level With Triggers, Andrew Morgan](https://sched.co/FmAJ){:target="_blank" rel="noopener noreferrer"})
+> Write less code and build apps faster! 
+
+Stitch is the 'Serverless platform from MongoDB', and it comes with a free tier to play around!
+Stitch provides a very easy way to implement new functionality without having to write lots of code in a separate backend application.
+The functionalities of Stitch are provided through an SDK.
```
Comment (TimDG) [inline _posts/2018-11-20-mongodb-europe-018.md]: 3rd Stitch and 2nd functionality in a row. Try to rewrite this sentence and avoid using these words. Since both this and the next one are quite short, perhaps combine them into 1?

File: _posts/2018-11-20-mongodb-europe-018.md [hunk 35]
```diff
@@ -0,0 +1,411 @@
...
+
+Stitch triggers combines change streams with Stitch Functions.
+So when the inventory of an article goes up, Stitch Trigger calls a function that uses Twilio to send a text message to your client.
+
+### Stitch Mobile Sync 
+Since 4.0, MongoDB provides a [Mobile](https://www.mongodb.com/products/mobile){:target="_blank" rel="noopener noreferrer"} version for IOS and Android.
+With Stitch, you can sync your data in your mobile application with your database.
+So now you can use the full MongoDB Query Language, including aggregations, on your mobile device and sync it with your database.
+
+### Build in external Integrations
```
Comment (TimDG) [inline _posts/2018-11-20-mongodb-europe-018.md]:
```suggestion
### Built-in external Integrations
```

File: _posts/2018-11-20-mongodb-europe-018.md [hunk 36]
```diff
@@ -0,0 +1,411 @@
...
+The fun with Stitch really starts when you combine all the goodness of Stitch with its integrations with cloud services like Twilio, AWS, Google, etc...
+You can authenticate with Google, store files on S3 or spin up a cluster on Redshift after you send a text message with Twilio.
+All of this can be hidden behind a simple function call for your application, or a trigger on your Atlas cluster.
+
+### Limited to Stitch UI?
+Luckily MongoDB builds its products with developers in mind.
+So you can import and export your Stitch applications and put them in a source control of your choice.
+
+# Meet the experts
```
Comment (TimDG) [inline _posts/2018-11-20-mongodb-europe-018.md]: Not really sure what this section adds

File: _posts/2018-11-20-mongodb-europe-018.md [hunk 37]
```diff
@@ -0,0 +1,411 @@
...
+    "password" : "secret",
+    "server.name" : "db-01.example.com",
+    "server.port" : "1433"
+  }
+}
+```
+
+They use Avro for the schema definition in combination with a schema-registry.
+Interested clients can then read the data off the topics and do their single-view-thing on the data and save it to MongoDB.
+The view is being build up, message per message.
```
Comment (TimDG) [inline _posts/2018-11-20-mongodb-europe-018.md]:
```suggestion
The view is being built up, message per message.
```

File: _posts/2018-11-20-mongodb-europe-018.md [hunk 38]
```diff
@@ -0,0 +1,411 @@
...
+At the conference you had the chance to book a 20 minute session with a MongoDB expert.
+This was of great help in getting to know the new MongoDB Aggregation Pipeline builder.
+The expert also gave some more tips in "thinking noSQL".
+
+> When data is shown together it is stored together -- MongoDB expert 
+
+> Data should be stored in the same way it is used -- MongoDB expert
+
+# [Streaming data pipelines with MongoDB and Kafka at AO](https://sched.co/FmAp){:target="_blank" rel="noopener noreferrer"}
+[AO](https://ao.com/){:target="_blank" rel="noopener noreferrer"} wanted to solve the issue of having data locked in different places so they wanted a [Single Customer View](https://en.wikipedia.org/wiki/Single_customer_view/){:target="_blank" rel="noopener noreferrer"}.
```
Comment (TimDG) [inline _posts/2018-11-20-mongodb-europe-018.md]: Who/what is AO? Perhaps an intro like in the next section?

File: _posts/2018-11-20-mongodb-europe-018.md [hunk 39]
```diff
@@ -0,0 +1,411 @@
...
+[Amadeus](https://amadeus.com/en) is the world's largest technology company dedicated to the travel industry.
+They have developed an impressive MongoDB farm, a large environment with 100 clusters.
+Some of these clusters run more than 100 shards, while others have 100TB MongoDB databases.
+Amadeus processes more than 1 trillion flight availability requests per day. 
+For each single search you do on a website they receive 500.000 availability requests.
+So search responsibly ;-)
+The number of requests per day grows by 50% each year.
+The second of January has by far the most requests, due to new years resolutions!
+If this day is in the weekend all systems are pushed to their limits.
+The fair database for one of their big clients, Kayak, is 100TB in size and changes daily.
```
Comment (TimDG) [inline _posts/2018-11-20-mongodb-europe-018.md]: fair database?

File: _posts/2018-11-20-mongodb-europe-018.md [hunk 40]
```diff
@@ -0,0 +1,411 @@
...
+
+I kid you not, that's it.
+
+Scale out or back in with a simple change in the config yaml and 
+```bash
+cubectl apply -f file.yaml
+```
+
+# MongoDB Charts
+- [Bringing Data to Life with MongoDB Charts](https://sched.co/FtWP), Guillaume Meister.
```
Comment (TimDG) [inline _posts/2018-11-20-mongodb-europe-018.md]: Why is this title so different from the others?

File: _posts/2018-11-20-mongodb-europe-018.md [hunk 41]
```diff
@@ -0,0 +1,411 @@
...
+
+Currently, if you want to visualize data in a MongoDB database you either have to code it yourself, or rely on a 3rd party tool and migrate your data to a different platform (for example: Kibana with Elasticsearch is a popular tool).
```
Comment (TimDG) [inline _posts/2018-11-20-mongodb-europe-018.md]:
```suggestion
Currently, if you want to visualize data in a MongoDB database you either have to code it yourself, or rely on a 3rd party tool and migrate your data to a different platform (for example: Kibana with Elasticsearch is very popular).
```

File: _posts/2018-11-20-mongodb-europe-018.md [hunk 42]
```diff
@@ -0,0 +1,411 @@
...
+Needless to say that this can be quite cumbersome.
+[MongoDB Charts](https://www.mongodb.com/products/charts) intends to solve this.
+
+So what is it? 
+MongoDB Charts is a visualization tool that runs as a standalone webserver, you access it via a web browser.
```
Comment (TimDG) [inline _posts/2018-11-20-mongodb-europe-018.md]: Isn't it rather redundant that you can access a standalone web server through a web browser?

File: _posts/2018-11-20-mongodb-europe-018.md [hunk 43]
```diff
@@ -0,0 +1,411 @@
...
+
+<div style="text-align: center;">
+  <img src="/img/2018-11-08-mongodb-europe-2018/mongodb-charts-01.gif" width="100%">
+</div>
+
+Charts is still in beta but you can already try it out. 
+MongoDB provides a docker image that you can [download](https://www.mongodb.com/download-center/charts) via the MongoDB download center.
+
+# Closing
+A great day being submerged in Mongo-knowledge.
```
Comment (TimDG) [inline _posts/2018-11-20-mongodb-europe-018.md]: Make this into a whole sentence

File: _posts/2018-11-20-mongodb-europe-018.md [hunk 44]
```diff
@@ -0,0 +1,411 @@
...
+</div>
+
+Charts is still in beta but you can already try it out. 
+MongoDB provides a docker image that you can [download](https://www.mongodb.com/download-center/charts) via the MongoDB download center.
+
+# Closing
+A great day being submerged in Mongo-knowledge. 
+This conference gave us plenty of opportunity to talk to other experts and learn about the new and upcoming features.  
+
+Exciting, we'll keep mongo-ing for quite a while!
```
Comment (TimDG) [inline _posts/2018-11-20-mongodb-europe-018.md]: Same here, make it a full sentence.

File: _posts/2018-11-20-mongodb-europe-018.md (hunk 25 above)
Comment (ChrisDeBruyne) [reply _posts/2018-11-20-mongodb-europe-018.md]: tom said no to Oxford Comma

File: _posts/2018-11-20-mongodb-europe-018.md (hunk 29 above)
Comment (ChrisDeBruyne) [reply _posts/2018-11-20-mongodb-europe-018.md]: Ik verkies de term "bewustmaking"
Veel nieuwelingen kennen deze site niet en het is zeker de moeite om het te vernoemen.

File: _posts/2018-11-20-mongodb-europe-018.md (hunk 28 above)
Comment (ChrisDeBruyne) [reply _posts/2018-11-20-mongodb-europe-018.md]: because it was in the Keynote but can be left out, it looks different from the other titles

File: _posts/2018-11-20-mongodb-europe-018.md (hunk 33 above)
Comment (ChrisDeBruyne) [reply _posts/2018-11-20-mongodb-europe-018.md]: It depends on the indexes you created.
It should point out that you should create an index specifically for your query.

File: _posts/2018-11-20-mongodb-europe-018.md (hunk 36 above)
Comment (ChrisDeBruyne) [reply _posts/2018-11-20-mongodb-europe-018.md]: Don't know about other vendor specific conferences but here they provide you the chance to talk to MongoDB engineers about your MongoDB related questions.

File: _posts/2018-11-20-mongodb-europe-018.md (hunk 33 above)
Comment (jvanderveken-ordina) [reply _posts/2018-11-20-mongodb-europe-018.md]: > Why? Or is this common knowledge?

It's an empirical finding, a statement of fact.

The "why" is expalined in the solution: $and and $or need different supporting indices.
